cachito/workers/tasks/rubygems.py

```diff
@@ -25,6 +41,70 @@ def cleanup_rubygems_request(request_id):
     nexus.execute_script("rubygems_cleanup", payload)
 
 
+@app.task
+@runs_if_request_in_progress
+def fetch_rubygems_source(request_id, package_configs=None):
+    """
+    Resolve and fetch RubyGems dependencies for a given request.
+
+    :param int request_id: the Cachito request ID this is for
+    :param list package_configs: the list of optional package configurations submitted by the user
+    """
+    validate_rubygems_config()
+    bundle_dir: RequestBundleDir = RequestBundleDir(request_id)
+
+    log.info("Configuring Nexus for RubyGems for the request %d", request_id)
+    set_request_state(request_id, "in_progress", "Configuring Nexus for RubyGems")
+    rubygems_repo_name = get_rubygems_hosted_repo_name(request_id)
+    prepare_nexus_for_rubygems_request(rubygems_repo_name)
+
+    log.info("Fetching dependencies for request %d", request_id)
+    package_configs = package_configs or [{}]
+    packages_data = []
+    for pkg_cfg in package_configs:
+        pkg_path = pkg_cfg.get("path", ".")
+        package_source_dir = bundle_dir.app_subpath(pkg_path).source_dir
+        set_request_state(
+            request_id,
+            "in_progress",
+            f"Fetching dependencies at the {pkg_path!r} directory",
+        )
+        request = get_request(request_id)
+        pkg_and_deps_info = resolve_rubygems(
+            package_source_dir,
+            request,
+        )
+
+        # defer DB operations to use the Nexus password in the env vars
+        packages_data.append(pkg_and_deps_info)
```

Comment:
This comment was in pip implementation and I don't really understand why. Is it pip specific or should I keep it here? Also, there was some logic about CA certificate, is that something I should implement here as well?